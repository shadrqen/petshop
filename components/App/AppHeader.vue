<template>
  <div>
    <v-toolbar
      :color="headerColor"
      class="elevation-0"
    >
      <router-link
        to="/"
        class="text-decoration-none white--text"
      >
        <base-avatar
          image-alt="Logo"
          image-src="headerIcon.PNG"
          :avatar-size="25"
        />
        petson.
      </router-link>
      <v-spacer />
      <base-button
        :text="true"
        btn-class="white--text"
      >
        <template #button-body>
          Products
          <v-icon>mdi-chevron-down</v-icon>
        </template>
      </base-button>
      <base-button
        :text="true"
        btn-class="white--text"
      >
        <template #button-body>
          Promotions
        </template>
      </base-button>
      <v-spacer />
      <base-button
        :outlined="true"
        btn-class="white--text"
      >
        <template #button-body>
          Cart ({{ cartItemsCount }})
        </template>
      </base-button>
      <client-only>
        <template v-if="userLoggedIn">
          <base-button
            btn-class="white--text mx-1"
            :outlined="true"
            @submitContent="openCloseLoginDialog(false)"
          >
            <template #button-body>
              Logout
            </template>
          </base-button>
          <base-avatar
            image-alt="Profile"
            image-src="profile.jpg"
            :avatar-size="35"
            avatar-class="mx-2"
          />
        </template>
        <template v-else>
          <base-button
            btn-class="white--text mx-1"
            :outlined="true"
            @submitContent="openCloseDialogs('login', 'open')"
          >
            <template #button-body>
              Login
            </template>
          </base-button>
        </template>
      </client-only>
    </v-toolbar>
    <login-dialog />
    <registration-dialog />
    <user-setting />
  </div>
</template>

<script>
import { mapState } from 'vuex'
import BaseButton from '@/components/Base/BaseButton'
import BaseAvatar from '@/components/Base/BaseAvatar'
import LoginDialog from '@/components/LoginDialog'
import RegistrationDialog from '@/components/RegistrationDialog'
import UserSetting from '@/components/UserSetting'
import DialogsMixin from '@/mixins/DialogsMixin'

export default {
  name: 'BaseHeader',
  components: {
    BaseAvatar,
    BaseButton,
    LoginDialog,
    RegistrationDialog,
    UserSetting
  },
  mixins: [DialogsMixin],
  data () {
    return {
      cartItemsCount: 0,
      headerColor: '#50c494'
    }
  },
  computed: {
    ...mapState('auth', ['userLoggedIn'])
  }
}
</script>

<style lang="scss" scoped>
</style>
