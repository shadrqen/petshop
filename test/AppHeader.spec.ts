import AppHeader from '~/components/App/AppHeader.vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'

import Vuex, {
  Store
} from 'vuex'

import {
  RootState,
  AuthState
} from '@/types/vuex'

describe('@/components/App/AppHeader.vue', () => {
  const localVue = createLocalVue()

  localVue.use(Vuex)

  const createDefaultState = (userLoggedIn: boolean = false): AuthState => ({
    userLoggedIn,
    loginDialog: false,
    logoutDialog: false,
    registrationDialog: false,
    accessToken: ''
  })

  const createStore = (
    userLoggedIn: boolean = false, state: AuthState = createDefaultState(userLoggedIn)
  ): Store<RootState> => new Vuex.Store({
    modules: {
      auth: {
        namespaced: true,
        state: {
          ...state
        }
      }
    }
  })

  const createWrapper = (
    userLoggedIn: boolean = false, store: Store<RootState> = createStore(userLoggedIn)
  ) => shallowMount(
    AppHeader,
    {
      localVue,
      store
    }
  )

  it('should render component and match snapshot', () => {
    const wrapper = createWrapper()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render app name on header', () => {
    const { element } = createWrapper()

    expect(element.innerHTML).toContain('petson.')
  })

  // it('should render login button when user is not logged in', () => {
  //   const { element } = createWrapper()
  //
  //   expect(element.innerHTML).toContain('Login')
  // })

  // it('should render logout button when user is logged in', () => {
  //   const { element } = createWrapper(true)
  //
  //   expect(element.innerHTML).toContain('Logout')
  // })
})
