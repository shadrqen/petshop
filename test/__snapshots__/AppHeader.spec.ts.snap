// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/App/AppHeader.vue should render component and match snapshot 1`] = `
"<div>
  <v-toolbar-stub color=\\"#50c494\\" tag=\\"header\\" extensionheight=\\"48\\" src=\\"\\" class=\\"elevation-0\\">
    <router-link to=\\"/\\" class=\\"text-decoration-none white--text\\">
      <base-avatar-stub avatarsize=\\"25\\" avatarclass=\\"\\" imagesrc=\\"headerIcon.PNG\\" imagealt=\\"Logo\\"></base-avatar-stub>
      petson.
    </router-link>
    <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
    <base-button-stub btnid=\\"\\" btnclass=\\"white--text\\" datatestid=\\"\\" text=\\"true\\" width=\\"auto\\" color=\\"\\" backgroundcolor=\\"\\" texttransform=\\"\\"></base-button-stub>
    <base-button-stub btnid=\\"\\" btnclass=\\"white--text\\" datatestid=\\"\\" text=\\"true\\" width=\\"auto\\" color=\\"\\" backgroundcolor=\\"\\" texttransform=\\"\\"></base-button-stub>
    <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
    <base-button-stub btnid=\\"\\" btnclass=\\"white--text\\" datatestid=\\"\\" outlined=\\"true\\" width=\\"auto\\" color=\\"\\" backgroundcolor=\\"\\" texttransform=\\"\\"></base-button-stub>
    <client-only>
      <base-button-stub btnid=\\"\\" btnclass=\\"white--text mx-1\\" datatestid=\\"\\" outlined=\\"true\\" width=\\"auto\\" color=\\"\\" backgroundcolor=\\"\\" texttransform=\\"\\"></base-button-stub>
    </client-only>
  </v-toolbar-stub>
  <login-dialog-stub></login-dialog-stub>
  <logout-notification-stub></logout-notification-stub>
  <registration-dialog-stub></registration-dialog-stub>
  <user-setting-stub></user-setting-stub>
</div>"
`;
